java/velox4j/src/main/cpp/main/velox4j/jni/JniWrapper.cpp

```diff
@@ -0,0 +1,471 @@
+      "Serde Memory Pool", memory::MemoryPool::Kind::kLeaf);
+  auto aggregationNode = std::const_pointer_cast<core::AggregationNode>(
+      ISerializable::deserialize<core::AggregationNode>(dynamic, serdePool));
+  auto type = exec::TableWriteTraits::outputType(aggregationNode);
+  auto serializedDynamic = type->serialize();
+  auto typeJson = folly::toPrettyJson(serializedDynamic);
+  return env->NewStringUTF(typeJson.data());
+  JNI_METHOD_END(nullptr)
+}
+
+jlong iSerializableAsCpp(JNIEnv* env, jobject javaThis, jstring json) {
+  JNI_METHOD_START
+  auto session = sessionOf(env, javaThis);
+  auto serdePool = session->memoryManager()->getVeloxPool(
+      "Serde Memory Pool", memory::MemoryPool::Kind::kLeaf);
+  spotify::jni::JavaString jJson{env, json};
+  auto dynamic = folly::parseJson(jJson.get());
+  auto deserialized = std::const_pointer_cast<ISerializable>(
+      ISerializable::deserialize<ISerializable>(dynamic, serdePool));
+  return session->objectStore()->save(deserialized);
+  JNI_METHOD_END(-1)
+}
+
+jlong variantAsCpp(JNIEnv* env, jobject javaThis, jstring json) {
+  JNI_METHOD_START
+  auto session = sessionOf(env, javaThis);
+  spotify::jni::JavaString jJson{env, json};
+  auto dynamic = folly::parseJson(jJson.get());
+  auto deserialized = variant::create(dynamic);
+  return session->objectStore()->save(std::make_shared<variant>(deserialized));
+  JNI_METHOD_END(-1)
+}
+
+class ExternalStreamAsUpIterator : public UpIterator {
+ public:
+  explicit ExternalStreamAsUpIterator(const std::shared_ptr<ExternalStream>& es)
+      : es_(es) {}
+
+  State advance() override {
+    VELOX_CHECK_NULL(pending_);
+    ContinueFuture future = ContinueFuture::makeEmpty();
+    auto out = es_->read(future);
+    if (out == std::nullopt) {
+      VELOX_CHECK(future.valid());
+      // Do not wait for the future to be fulfilled, just return.
+      return State::BLOCKED;
+    }
+    VELOX_CHECK(!future.valid());
+    if (out == nullptr) {
+      return State::FINISHED;
+    }
+    pending_ = out.value();
+    return State::AVAILABLE;
+  }
+
+  void wait() override {
+    VELOX_CHECK_NULL(pending_);
+    VELOX_NYI("Not implemented: {}", __func__);
+  }
+
+  RowVectorPtr get() override {
+    VELOX_CHECK_NOT_NULL(
+        pending_,
+        "ExternalStreamAsUpIterator: No pending row vector to return. Make "
+        "sure the iterator is available via member function advance() first");
+    auto out = pending_;
+    pending_ = nullptr;
+    return out;
+  }
+
+ private:
+  const std::shared_ptr<ExternalStream> es_;
+  RowVectorPtr pending_{nullptr};
+};
+
+jlong createUpIteratorWithExternalStream(
+    JNIEnv* env,
+    jobject javaThis,
+    jlong id) {
+  JNI_METHOD_START
+  auto es = ObjectStore::retrieve<ExternalStream>(id);
+  return sessionOf(env, javaThis)
+      ->objectStore()
+      ->save(std::make_shared<ExternalStreamAsUpIterator>(es));
+  JNI_METHOD_END(-1L)
+}
+} // namespace
+
+void JniWrapper::mapFields() {}
```

Comment:
Do we have any TODO for this empty function?

Reply:
No need to add todos. They are just empty implementations. I added some comments.